comments_index.py

```diff
@@ -0,0 +1,94 @@
+import logging
+
+from nltk.stem.snowball import SnowballStemmer
+from nltk.tokenize import TweetTokenizer
+
+from helper.csv_helper import CsvHelper
+from helper.index_file_helper import IndexFileHelper
+from helper.model_helper import CommentsHelper
+from models.comment import Comment
+from models.index import Index
+
+
+class CommentsIndex:
+    def __init__(self, rebuild_index=False):
+        self.index = Index(CommentsHelper().get_latest_file())
+        self.stemmer = SnowballStemmer("english")
+        self.tokenizer = TweetTokenizer()
+        self.helper = IndexFileHelper()
+        if rebuild_index:
+            self.build_index()
+        else:
+            self.restore_index()
+
+    def __del__(self):
+        self.helper.write_index(self.index)
+
+    def restore_index(self):
+        try:
+            self.index = self.helper.get_index()
+            if self.index.get_file() != CommentsHelper().get_latest_file():
+                logging.debug(
+                    'Restored index based on {}. To create the index of last file use attribute "index".'.format(
+                        self.index.get_file()))
+        except FileNotFoundError:
+            logging.info('No previous index was found. Creating new one.')
+            self.build_index()
+
+    def search(self, query: str, number_of_results: int):
+        if len(query.split(' ')) > 1:
+            raise AttributeError('Only single word query\'s are supported for now')
+        processed_query = self.get_tokens(query)
+        results = list()
+        for token in processed_query:
+            posting_pointer = self.index.get(token)
+            if not posting_pointer:
+                continue
+            i = 0
+            for posting in self.helper.get_posting(posting_pointer[0], posting_pointer[1]):
```

Comment:
you are saving int value in the index so this line failes with: TypeError: 'int' object is not subscriptable